sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateBoxDao.java

```diff
@@ -182,6 +184,61 @@ public BoxUse getUseById(long id) throws IOException {
     return results;
   }
 
+  @Override
+  public List<Box> getByPartialSearch(String search) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)